src/Components/News/NewsListItem.js

```diff
@@ -1,31 +1,45 @@
-import React from 'react';
+import React, { Component } from 'react';
+import PageContext from '../../PageContext';
 import PropTypes from 'prop-types';
 import { Card, CardImg, CardTitle, CardText, CardSubtitle, CardBody } from 'reactstrap';
 import { Link } from 'react-router-dom';
 import DateFormat from '../Shared/DateFormat';
 
 import IMAGE_PLACEHOLDER from '../../theme/images/placeholder.jpg';
 
-const ItemLink = (props) => {
-    return (
-        <Link
-            to={{
-                pathname: props.pathname,
-                state: props.state
-            }}
-        >
-            {props.children}
-        </Link>
-    );
-};
+class ItemLink extends Component {
+    static contextType = PageContext;
+    languagePrefix = '';
+    render() {
+        this.languagePrefix =
+            this.context.language && this.context.language.code
```

Comment:
We're doing this same thing in 3 different component, we need to encapsulate this and be as DRY as possible. We might need a [HOC](https://reactjs.org/docs/higher-order-components.html) to do this.